docs/launch-with/galaxy-launch-with.rst

```diff
@@ -4,23 +4,30 @@ Galaxy
 
 Dockstore integrates with the `Galaxy Application <https://usegalaxy.org>`__, allowing you to launch Galaxy-based workflows from Dockstore into
 any Galaxy instance. This works both from within the Dockstore interface and from within the Galaxy interface.
-Here is some information on what that looks like from a user point of view in a mini tutorial.
+There are several ways to launch Galaxy-based workflows. Here is some information on what that looks like from a user point of view in a mini tutorial.
 
 While browsing Dockstore
 ========================
-When browsing Galaxy workflows from within Dockstore, you will see a
-Launch with Galaxy button on the right.
+
+Dockstore has a fully featured faceted search to effectively search for workflows. On the left panel, you can choose from several options to narrow down your search. Searching for workflows can be done using a specific search term. Results can be narrowed down based on the language of the workflow, author, organization, source control used among others.
+
+Once you have clicked on a workflow, you will see a Launch with Galaxy button on the right.
 
 .. figure:: /assets/images/docs/galaxy/galaxy_workflow.png
    :alt: Galaxy workflow on Dockstore
 
-The button will bring up a dropdown window where you may select which Galaxy instance to launch the workflow into. A few default options will
-be provided by Dockstore, but if you need to launch into another instance, you may provide a custom URL. For your custom URL to work properly,
+The button will bring up a dropdown window where you may select which Galaxy instance to launch the workflow into. A few default options are provided by Dockstore, but if you need to launch into another instance, you may provide a custom URL. For your custom URL to work properly,
 it must start with ``https://`` and there should not be an ending forward slash ``/``.
 
 .. figure:: /assets/images/docs/galaxy/launch_with_options.png
    :alt: Launch options for Galaxy
+   
+You can use Terra to spin up a custom cloud environment for Galaxy. To do that, navigate to the Notebooks tab on any Workspace on `Terra <https://app.terra.bio/>`__ and click on the Create a Cloud Environment for Galaxy on the left.
```

Comment:
Appearance of Terra seems a bit abrupt since we introduce usegalaxy.org on line 5. 
Thinking this needs a header like "While browsing Galaxy", for example "While using Terra"
and should be moved below "While browsing Galaxy" to complete the thought about usegalaxy.org before moving onto Terra.

Reply:
The placement of this is a bit tricky. What I was going for is that, on Dockstore, user can use the default cloud options or spin up a custom cloud environment on Terra (BDC or AnVIL included). And hence it is under the While browsing Dockstore header. The concept was to introduce 3 different ways to launch the Galaxy workflows:

1. From Dockstore, use default cloud instance or add unique Galaxy cloud instance URL (up to and including /galaxy)
2. Within Galaxy, hit import, use their search feature
3. Within Galaxy, hit import, copy TRS ID from Dockstore entry

But if the 1st way is elaborate and the introduction of Terra is abrupt, we can technically introduce that as a different way to launch Galaxy. So we can have browsing Dockstore and using the available cloud instances as one way. And introduce Terra and how to create a custom cloud instance and paste it in Dockstore as a whole other way.

I'd love to know your thoughts! @denis-yuen and @coverbeck @bethsheets

Reply:
> using Galaxy with controlled access data. It is my impression that the other available Galaxy instances are free, but users can only bring small and open datasets, and users are limited in computing resources.

Hmm, I'm not so sure about that. I infer, maybe incorrectly, that many institutions run their own Galaxy instances, and I would think those have access to controlled access data.

And we saw Galaxy is used by Australian BioCommons, which I assume will have controlled access data.

Reply:
This is true, Charles. I think we can avoid discussing the nuance in the documentation. But to clarify, my discussion around controlled access data relates to non-members of these Galaxy servers using them with controlled access data. Generally, controlled data access requests require the researcher to work with the compute clusters to follow specific practices. It is difficult to obtain this when using a free resource that your sponsoring institute is not affiliated with.